install/nodejs

```diff
@@ -0,0 +1,275 @@
+
+        if [[ -e "${check_file}" && ! -L "${check_file}" ]]; then
+            ewarn "Previous node configuration found! ${check_file}"
+            die "${check_file} does not appear to be a symlink, failing!"
+        fi
+        edebug "Removing previous symlink to ${check_file}"
+        rm -f "${check_file}"
+    done
+
+    # Remove existing dir symlink as it will be replaced by current lts
+    edebug "Removing previous node symlink: ${install_prefix}/node -> ${orig_install}"
+    rm -f "${install_prefix}/node"
+
+    eprogress_kill
+}
+
+
+opt_usage retrieve_sha <<'END'
+Retrieves nodejs release sha for the lts codename
+END
+retrieve_sha()
+{
+    $(opt_parse                                                    \
+        ":lts_codename     c | The codename of the lts to install" \
+        ":sha_output_path  o | The place to store the sha"         \
+    )
+
+    einfo "Retrieving SHASUM256.txt for codename ${lts_codename}"
+    local sha_url="https://nodejs.org/download/release/latest-${lts_codename}/SHASUMS256.txt"
+    edebug "Using url: ${sha_url}"
+
+    # Download SHA256 to determine version
+    eretry efetch --style=einfo "${sha_url}" "${sha_output_path}" |& edebug
+}
+
+opt_usage get_latest_node_version <<'END'
+Retrieves the latest semantic version for a given nodejs codename
+END
+get_latest_node_version()
+{
+    $(opt_parse                                                                    \
+        ":lts_codename    c | The lts codename to retrieve the latest version for" \
+        ":sha_output_path o | The path where the sha should be saved"
+    )
+
+    # Retrieve the sha file and apply cleanup
+    retrieve_sha --lts-codename "${lts_codename}" --sha-output-path "${sha_output_path}"
+    trap_add "rm -f ${sha_output_path}"
+
+    edebug "$(cat "${sha_output_path}")"
+    grep "node-v" "${sha_output_path}" | head -n 1 | awk -F- '{ print $2 }' | sed 's/v//g'
+}
+
+opt_usage install_node <<'END'
+Installs the nodejs lts version based on the codename passed in
+END
+install_nodejs()
+{
+    $(opt_parse                                                           \
+        ":install_prefix p | The install prefix"                          \
+        ":lts_codename   c | The codename for the lts version to install"
+    )
+
+    lts_version=$(get_latest_node_version --lts-codename "${lts_codename}" --sha-output-path "${default_sha_path}")
+    einfo "Installing latest version of nodejs codename '${lts_codename}': ${lts_version}"
+
+    if os darwin; then
+        package=""
+
+        mac_lts_version=$(cut -d. -f1 <<< "${lts_version}")
+        package="node@${mac_lts_version}"
+        edebug "Mac LTS package to install: ${package}"
+
+        # Install the latest codenamed version of node using brew
+        pkg_install --sync "node@${mac_lts_version}"
+
+        # Workaround issue with Catalina
+        command_exists "npm" || brew link "node@${mac_lts_version}"
```

Comment:
I spun up a Catalina VM to replicate an issue the pipeline exposed. Apparently, the `node@16` package isn't symlinking the packages binaries. This was just a _quick_ fix that seemed safe whether upstream is fixed or not. I wasn't immediately able to see what the issue was with upstream, or at least why it impacts, seemingly, only Catalina.

@marshall-mcmullen thoughts on this?

Reply:
I am not sure why it wouldn't work on Catalina the same. I'm guessing a bug in brew onolder Mac perhaps? I think your workaround here is great.